src/components/ui/Icon/README.md

```diff
@@ -0,0 +1,188 @@
+- Communicating a link, button, or action. Use a `Button` or `Tab` (combined with an `Icon`) instead.
+
+## Appearance
+
+### Colour
+
+An icon can be passed either a stroke or a fill colour.
+
+```jsx
+<Icon name="pen-tool" strokeColor="#053e6c" large/>
+<Icon name="target" strokeColor="#8a0000" large/>
+<Icon name="star" strokeColor="#eeb600" large/>
+<br />
+<Icon name="pen-tool" fillColor="#053e6c" large/>
+<Icon name="target" fillColor="#8a0000" large/>
+<Icon name="star" fillColor="#eeb600" large/>
+<br />
+<Icon name="pen-tool" fillColor="#5788b1" strokeColor="#043258" large/>
+<Icon name="target" fillColor="#c26767" strokeColor="#780000" large/>
+<Icon name="star" fillColor="#f7e095" strokeColor="#d7a400" large/>
+```
+
+When pairing an icon with a label, using a slightly lighter variant of the text colour helps make the two elements feel connected, whilst ensuring the icon doesn't have more visual weight than the text. Nautilus does this by default because we always want you to look good. 🙌
+
+```jsx
+import { Paragraph } from '@octopusthink/nautilus';
+
+<React.Fragment>
+  <Paragraph style={{ color: '#b32973' }}>
+    <Icon name="heart"/> I love Lucy
+  </Paragraph>
+  <Paragraph>
+    <Icon name="coffee" />
+    Lucy loves coffee
+  </Paragraph>
+</React.Fragment>
+```
+
+### Size
+
+Icons are available in five different sizes: `extraLarge` (`64rem`), `large` (`48rem`), `medium` (`24px`), `small` (`16px`), and `extraSmall` (`8px`). If you don't pass a size value, your icons will use the medium size.
+
+The default icon sizes can be tweaked in your theme. Sizes should be defined in `rem` to ensure the icons adjust to the user's font size. (The above sizes are the pixel equivalents of the `rem` sizes provided for reference.)
+
+```jsx
+<Icon name="heart" extraSmall />
+<Icon name="heart" small />
+<Icon name="heart" medium />
+<Icon name="heart" large />
+<Icon name="heart" extraLarge />
+
+<Icon name="star" extraSmall />
+<Icon name="star" small />
+<Icon name="star" medium />
+<Icon name="star" large />
+<Icon name="star" extraLarge />
+
+<Icon name="pen-tool" extraSmall />
+<Icon name="pen-tool" small />
+<Icon name="pen-tool" medium />
+<Icon name="pen-tool" large />
+<Icon name="pen-tool" extraLarge />
+```
+
+Stroke widths scale with the icon size for better legibility. Stroke widths are also customisable via your theme.
+
+[example]
+
+### Margin & alignment
+
+By default, an `Icon` has a small amount of margin to its right, to give it some breathing space when used with text.
+
+You can customise the margin value to ensure proper placement in your layout. To maintain visual consistency, it's best to use the spacing values provided by your theme.
+
+Aligning an icon perfectly to text it's inline with can sometimes be tricky. By default, the container is set to vertical-align middle, but you can override this if needed.
+
+```jsx
+import { Paragraph } from '@octopusthink/nautilus';
+
+<React.Fragment>
```

Comment:
I'm really starting to doubt this part; this is easily done with a `css` prop if you're using emotion, or with regular styles + `className` if you aren't. We're just putting these values right into our styles, and I don't think it's a good API.

We should provide defaults and explain them, but I don't think this is an API woth supporting. I get it's a bit more effort to write `css={css`margin: 0;`} than `margin="0"`, but it's not something we'll need to do often, right? Or if it is someone can extend this and make their own `Icon` component that takes a `margin` prop. I'd really like to drop these props before we ship.

Reply:
The one exception to this I could see is if the `Icon` is used without any text; a `noMargin` prop could be handy, but I don't want to support this kind of API where we're just duplicating what (better, more flexible) solutions can provide.

Reply:
(These props affect the `span` wrapper, not the SVG itself, right? Or they could? I get that colouring the SVG is a bit more tedius because of our wrapper, but I don't think we need to provide an API for the margin.)

Reply:
Here are a few reasons why I think it's worth including what are essentially CSS props as part of the supported API:
- You can still go ahead and use CSS if you want to, so it becomes a more flexible solution.
- It encourages people _not_ to use CSS to style components, but rather use JS variables provided. This could also help to reinforce various design guidelines (ie, we don't provide an explicit prop for that CSS attribute, so maybe people will think twice about overriding the defaults with CSS. Ideally consumers of the design system would only modify the theme variables and add some layout-level CSS.
- It allows us to apply more programmatic rules, like colouring icon stroke differently depending on how fill colour is passed (see #92). 
- It makes it easier for users to quickly overwrite a property in a clean and easy way. Since we're optimising for API ease-of-use, this means we're following our own rules here—we're doing a bit of extra legwork at the API level to make it simpler at the user level.

Reply:
I should address these:

> You can still go ahead and use CSS if you want to, so it becomes a more flexible solution.
That means we're providing an extra way to do a thing we don't encourage in the first place. I get what you're saying about discouraging CSS usage, but if they get to the point of needing _custom_ values (which is what our `margin` prop would allow), they've got to the point of just using CSS anyway. I understand that we're providing an "easier" (though barely) way to style a component, but it's an easier way to do something you don't want to encourage.

From what I can tell the two _common_ scenarios are:

* use the default theme values you should be supplying with your theme
* use no margin because it's a solo icon

Anything else should be a rarity and it's fine to one-off it. Or maybe we should have multiple theme values. Overriding the defaults being discouraged is fine, but if that's the case why offer an override prop?

I should be clear: I'm talking about just the `margin` prop. For things like colour, which are pretty fundamental to an icon and are encouraged/likely to be changed by the user, it's fine to have a prop for it.

Reply:
Update: following some further in-person discussion, I think a compromise here is to make the margin prop a bit smarter and more restrictive, in accordance with actual usage patterns. Suggested an approach in #101, and for now I've nixed the margin prop (it's still themeable though!) and we can reassess once we've had an opportunity to better explore how this component actually works in practise.